src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +392,461 @@ else if (!period.isEmpty()) {
+    }
+
+    /**
+     * Checks if the string ends with period.
+     *
+     * @param sentence string to check for period at end.
+     * @param period String to check within sentence.
+     * @return true, is sentence ends with period.
+     */
+    private static boolean isPeriodAtEnd(String sentence, String period) {
+        final String summarySentence = sentence.trim();
+        return summarySentence.lastIndexOf(period) != summarySentence.length() - 1;
+    }
+
+    /**
+     * Checks if the summary is valid.
+     *
+     * @param ast Javadoc root node.
+     * @return true, if sentence is invalid.
+     */
+    private boolean isSummaryValid(DetailNode ast) {
+        final String inlineFirstSentence = getInlineSummary(ast);
+        final int endOfInlineSentence = inlineFirstSentence.lastIndexOf(period);
+        return containsForbiddenFragment(inlineFirstSentence.substring(0,
+                endOfInlineSentence))
+                || !containsCorrectInlineTags(ast)
+                || !validInlineTagInsideHtmlFormat(ast);
+    }
+
+    /**
+     * Finds and return if the summary sentence has allowed Inline Tags.
+     *
+     * @param ast Children of javadoc Inline Tag DetailNode[].
+     * @return true, if first sentence contains allowed tags.
+     */
+    private static boolean containsCorrectInlineTags(DetailNode ast) {
+        boolean contains = true;
+        for (DetailNode node : ast.getChildren()) {
+            final DetailNode[] child = node.getChildren();
+            if (child.length > 1
+                    && child[1].getType() == JavadocTokenTypes.CUSTOM_NAME
+                    && getInlineTagsInsideAnotherInlineTag(child) != null
+                    && !ALLOWED_INLINE_TAGS.contains(getInlineTagsInsideAnotherInlineTag(child)
+                        .getText())) {
+                contains = false;
+                break;
+            }
+        }
+        return contains;
+    }
+
+    /**
+     * Finds and returns only those inline tag nodes which are inside html tags from an ast node.
+     *
+     * @param ast Root node.
+     * @return DetailNode of inline tag.
+     */
+    private static DetailNode getInlineTagNodeInsideHtmlTagsOnly(DetailNode ast) {
+        DetailNode result = null;
+        for (DetailNode child: ast.getChildren()) {
+            if (child.getType() == JavadocTokenTypes.HTML_ELEMENT) {
+                result = getInlineTagNodeWithinHtmlElement(child);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Returns description node inside HTML type description.
+     *
+     * @param inlineTag Inline Tag node inside HTML.
+     * @return Description Node.
+     */
+    private static DetailNode getDescriptionNodeInsideHtml(DetailNode inlineTag) {
+        DetailNode node = null;
+        for (DetailNode child: inlineTag.getChildren()) {
+            if (child.getType() == JavadocTokenTypes.DESCRIPTION) {
+                node = child;
```

Comment:
should we do break ?

Reply:
There is always a single `DESCRIPTION` so it wouldn't matter here. Instead changed to `findFirstToken()`